File: drivers/Connector-cpp/CompareRs.cpp
/*
 * Copyright (C) 2016-2019 ActionTech.
 * License: https://www.mozilla.org/en-US/MPL/2.0 MPL version 2 or higher.
 */
#include <stdlib.h>
#include <iostream>
#include<string>
#include<list>
using namespace std;



bool compareList(list<string> dblerslist, list<string> mysqlrslist, bool allow_diff_sequence) {
	if (dblerslist.size() != mysqlrslist.size()) {
		//cout << "not equal" << endl;
		return false;
	}

	if (allow_diff_sequence) {
		dblerslist.sort();
		mysqlrslist.sort();
	}

	list<string>::iterator iter1;
	list<string>::iterator iter2;
	for (iter1 = dblerslist.begin(), iter2 = mysqlrslist.begin(); iter1 != dblerslist.end(); iter1++, iter2++) {
		if (*iter1 != *iter2) {
			//cout << "not equal" << endl;
			return false;
			//break;
		}
	}
	return true;
}

File: drivers/Connector-.Net/netdriver/comparers.cs
/*
 * Copyright (C) 2016-2022 ActionTech.
 * License: https://www.mozilla.org/en-US/MPL/2.0 MPL version 2 or higher.
 */
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace netdriver
{
    class CompareRs
    {
        public static bool CompareRS(List<String> dblers, List<String> mysqlrs,bool allow_diff)
        {
            if (allow_diff)
                return true;
            if (dblers == mysqlrs)
                return true;
            if (null == dblers && null == mysqlrs)
                return true;
            if (null == dblers || null == mysqlrs)
                return false;
            if (dblers.Count != mysqlrs.Count || !dblers.All(mysqlrs.Contains))
                return false;
//            if (allow_diff_sequence)
//            {
//                dblers.Sort();
//                mysqlrs.Sort();
//
//            }
            int nCount = dblers.Count;
            for (int n = 0; n < nCount; n++)
            {
                if (0 != string.Compare(dblers[n], mysqlrs[n], false))
                {
                    return false;
                }
            }
            return true;
        }
    }
}
